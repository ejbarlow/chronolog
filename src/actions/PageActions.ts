enum PageActions {
  ADD = '_PAGE_ACTION_ADD_',
  SUB = '_PAGE_ACTION_SUB_',
  SET = '_PAGE_ACTION_SET_',
  SOURCE = '_PAGE_ACTION_SOURCE_',
  SET_DATE = '_PAGE_ACTION_DATE_',
}

export { PageActions as default };
