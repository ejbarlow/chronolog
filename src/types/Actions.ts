export enum ActionType {
  PAGE_ADD = '_PAGE_ACTION_ADD_',
  PAGE_SUB = '_PAGE_ACTION_SUB_',
  PAGE_SET = '_PAGE_ACTION_SET_',
  DATE_SET = '_DATE_ACTION_SET_',
}

export type StateAction =
  | { type: ActionType.PAGE_ADD }
  | { type: ActionType.PAGE_SUB }
  | { type: ActionType.PAGE_SET, payload: number }
  | { type: ActionType.DATE_SET, payload: Date };
