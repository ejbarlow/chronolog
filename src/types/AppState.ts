import ScanProps from "./ScanProps";

type AppState = {
  scans: ScanProps[];
  manifestPath: string;
  volume: number;
  highestPage: number;
  page: number;
  date: Date;
};

export type { AppState as default };
