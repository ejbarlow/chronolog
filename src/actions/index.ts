import PageActions from './PageActions';

const GlobalActions = PageActions;

export { GlobalActions as default };
