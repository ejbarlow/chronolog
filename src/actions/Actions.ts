import { Action, ActionType } from "../types/Actions";

export const PAGE_ADD = (): ActionType => ({
  type: Action.PAGE_ADD,
});

export const PAGE_SUB = (): ActionType => ({
  type: Action.PAGE_SUB,
});

export const DATE_SET = (date: Date): ActionType => ({
  type: Action.DATE_SET,
  payload: date,
});

export const PAGE_SET = (pageNum: number): ActionType => ({
  type: Action.PAGE_SET,
  payload: pageNum,
});

export const SCANS_SRC = (path: string): ActionType => ({
  type: Action.SCANS_SRC,
  payload: path,
});
