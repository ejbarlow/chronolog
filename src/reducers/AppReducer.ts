import AppState from "../types/AppState";
import { Action, ActionType } from "../types/Actions";
import ScanProps from "../types/ScanProps";

const AppReducer = (state: AppState, action: ActionType): AppState => {
  let page;
  switch (action.type) {
    /**
     * Set, increment and decrement page number.
     */
    case Action.PAGE_SET:
      page = Math.min(Math.max(action.payload, 0), state.highestPage);
      return {
        ...state,
        page,
        currentScan: findScan(state.scans, page, state.volume, state.date),
      };

    case Action.PAGE_ADD:
      page = Math.min(state.page + 2, state.highestPage);
      return {
        ...state,
        page,
        currentScan: findScan(state.scans, page, state.volume, state.date),
      };

    case Action.PAGE_SUB:
      page = Math.max(state.page - 2, 0);
      return {
        ...state,
        page,
        currentScan: findScan(state.scans, page, state.volume, state.date),
      };

    /**
     * Set the volume number.
     */
    case Action.VOL_SET:
      const volScans = state.scans.filter(
        (scan) => scan.volume === action.payload
      );
      if (!volScans.length) return state;
      const volHighest = getHighestPage(volScans);
      const newPage = Math.min(volHighest, state.page);
      return {
        ...state,
        volume: action.payload,
        highestPage: volHighest,
        page: newPage,
        currentScan: findScan(volScans, newPage, action.payload, state.date),
      };

    /**
     * Set the date.
     */
    case Action.DATE_SET:
      return {
        ...state,
        date: action.payload,
        currentScan: findScan(
          state.scans,
          state.page,
          state.volume,
          action.payload
        ),
      };

    /**
     * Set the current scan to the previous/next one available for the
     * current page & volume (if one exists).
     */
    case Action.DATE_ADD:
    case Action.DATE_SUB:
      if (!state.currentScan) return state;
      const processedScans = state.scans
        .filter(
          (scan) =>
            scan.volume === state.volume && scan.pages.includes(state.page)
        )
        .sort((a, b) => (a.date > b.date ? 1 : -1));
      const nextIndex = Math.min(
        Math.max(
          action.type === Action.DATE_SUB
            ? processedScans.indexOf(state.currentScan) - 1
            : processedScans.indexOf(state.currentScan) + 1,
          0
        ),
        processedScans.length - 1
      );
      const nextScan = processedScans[nextIndex];
      return {
        ...state,
        date: nextScan.date,
        currentScan: nextScan,
      };

    /**
     * Import a set of scans.
     */
    case Action.SCANS_SRC:
      const scansHighest = getHighestPage(
        action.payload.filter((scan) => scan.volume === state.volume)
      );
      page = Math.min(Math.max(state.page, 0), scansHighest);
      return {
        ...state,
        scans: action.payload,
        highestPage: scansHighest,
        page,
        currentScan: findScan(action.payload, page, state.volume, state.date),
      };

    /**
     * Import a JSON manifest.
     */
    case Action.MANIFEST_SRC:
      return { ...state, manifestPath: action.payload };
    default:
      return state;
  }
};

/**
 * Used to find `currentScan` - the scan taken closest to `state.date` of all
 * versions of the current page and volume.
 *
 * @param scans An array containing all scans to be searched.
 * @param page Find scans containing this page number.
 * @param volume Find scans from this volume.
 * @param date Find the scan taken closest to this date.
 * @return ScanProps An array containing the most recent scan of each page,
 */
const findScan = (
  scans: ScanProps[],
  page: number,
  volume: number,
  date: Date
) => {
  const filteredScans = scans.filter(
    (scan) => scan.pages.includes(page) && scan.volume === volume
  );
  if (filteredScans.length < 1) {
    return undefined;
  }
  const newScan = filteredScans.reduce((closest, curr) => {
    return Math.abs(curr.date.getTime() - date.getTime()) <
      Math.abs(closest.date.getTime() - date.getTime()) &&
      curr.pages.includes(page)
      ? curr
      : closest;
  });
  return newScan;
};

/**
 * Get the highest page number present in a given set of scans.
 *
 * @param scans The scans to parse.
 * @return number The highest page number present.
 */
const getHighestPage = (scans: ScanProps[]) => {
  const max = Math.max(...scans.map((scan) => Math.max(...scan.pages)), 0);
  return max;
};

export { AppReducer as default };
