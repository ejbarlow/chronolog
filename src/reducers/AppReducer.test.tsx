import AppReducer from "./AppReducer";
import * as Action from "../actions/Actions";
import AppState from "../types/AppState";

/**
 * Initial app state.
 */
const mockInitialState: AppState = {
  scans: [],
  volume: 1,
  highestPage: 10,
  page: 0,
  date: new Date("Wed Jan 05 2022 19:16:33 GMT+0000 (Greenwich Mean Time)"),
  manifestPath: "/scans/scan-manifest.json",
};

const mockScans = [
  {
    uid: "CL_000000",
    path: "public/1/2022_01_01/JRNL_00-01_0001.webp",
    thumbPath: "public/1/2022_01_01/JRNL_00-01_0001_thumb.webp",
    volume: 1,
    pages: [0, 1],
    date: new Date("2022-01-01T00:00:00.000Z"),
    version: 0,
  },
  {
    uid: "CL_000001",
    path: "public/1/2022_01_01/JRNL_02-03_0001.webp",
    thumbPath: "public/1/2022_01_01/JRNL_02-03_0001_thumb.webp",
    volume: 1,
    pages: [2, 3],
    date: new Date("2022-01-01T00:00:00.000Z"),
    version: 0,
  },
  {
    uid: "CL_000002",
    path: "public/1/2022_01_01/JRNL_04-05_0001.webp",
    thumbPath: "public/1/2022_01_01/JRNL_04-05_0001_thumb.webp",
    volume: 1,
    pages: [4, 5],
    date: new Date("2022-01-01T00:00:00.000Z"),
    version: 0,
  },
  {
    uid: "CL_000003",
    path: "public/1/2022_01_01/JRNL_06-07_0001.webp",
    thumbPath: "public/1/2022_01_01/JRNL_06-07_0001_thumb.webp",
    volume: 1,
    pages: [6, 7],
    date: new Date("2022-01-01T00:00:00.000Z"),
    version: 0,
  },
  {
    uid: "CL_000004",
    path: "public/1/2022_01_01/JRNL_08-09_0001.webp",
    thumbPath: "public/1/2022_01_01/JRNL_08-09_0001_thumb.webp",
    volume: 1,
    pages: [8, 9],
    date: new Date("2022-01-01T00:00:00.000Z"),
    version: 0,
  },
  {
    uid: "CL_000005",
    path: "public/1/2022_01_01/JRNL_10-11_0001.webp",
    thumbPath: "public/1/2022_01_01/JRNL_10-11_0001_thumb.webp",
    volume: 1,
    pages: [10, 11],
    date: new Date("2022-01-01T00:00:00.000Z"),
    version: 0,
  },
  {
    uid: "CL_000006",
    path: "public/1/2022_01_02/JRNL_04-05_0001.webp",
    thumbPath: "public/1/2022_01_02/JRNL_04-05_0001_thumb.webp",
    volume: 1,
    pages: [4, 5],
    date: new Date("2022-01-02T00:00:00.000Z"),
    version: 0,
  },
  {
    uid: "CL_000007",
    path: "public/1/2022_01_02/JRNL_06-07_0001.webp",
    thumbPath: "public/1/2022_01_02/JRNL_06-07_0001_thumb.webp",
    volume: 1,
    pages: [6, 7],
    date: new Date("2022-01-02T00:00:00.000Z"),
    version: 0,
  },
  {
    uid: "CL_000008",
    path: "public/1/2022_01_02/JRNL_10-11_0001.webp",
    thumbPath: "public/1/2022_01_02/JRNL_10-11_0001_thumb.webp",
    volume: 1,
    pages: [10, 11],
    date: new Date("2022-01-02T00:00:00.000Z"),
    version: 0,
  },
  {
    uid: "CL_000009",
    path: "public/1/2022_01_02/JRNL_12-13_0001.webp",
    thumbPath: "public/1/2022_01_02/JRNL_12-13_0001_thumb.webp",
    volume: 1,
    pages: [12, 13],
    date: new Date("2022-01-02T00:00:00.000Z"),
    version: 0,
  },
  {
    uid: "CL_000010",
    path: "public/2/2022_01_02/JRNL_00-01_0001.webp",
    thumbPath: "public/2/2022_01_02/JRNL_00-01_0001_thumb.webp",
    volume: 2,
    pages: [0, 1],
    date: new Date("2022-01-02T00:00:00.000Z"),
    version: 0,
  },
  {
    uid: "CL_000011",
    path: "public/2/2022_01_02/JRNL_02-03_0001.webp",
    thumbPath: "public/2/2022_01_02/JRNL_02-03_0001_thumb.webp",
    volume: 2,
    pages: [2, 3],
    date: new Date("2022-01-02T00:00:00.000Z"),
    version: 0,
  },
  {
    uid: "CL_000012",
    path: "public/2/2022_01_02/JRNL_04-05_0001.webp",
    thumbPath: "public/2/2022_01_02/JRNL_04-05_0001_thumb.webp",
    volume: 2,
    pages: [4, 5],
    date: new Date("2022-01-02T00:00:00.000Z"),
    version: 0,
  },
  {
    uid: "CL_000013",
    path: "public/2/2022_01_02/JRNL_06-07_0001.webp",
    thumbPath: "public/2/2022_01_02/JRNL_06-07_0001_thumb.webp",
    volume: 2,
    pages: [6, 7],
    date: new Date("2022-01-02T00:00:00.000Z"),
    version: 0,
  },
  {
    uid: "CL_000014",
    path: "public/2/2022_01_02/JRNL_08-09_0001.webp",
    thumbPath: "public/2/2022_01_02/JRNL_08-09_0001_thumb.webp",
    volume: 2,
    pages: [8, 9],
    date: new Date("2022-01-02T00:00:00.000Z"),
    version: 0,
  },
  {
    uid: "CL_000015",
    path: "public/2/2022_01_03/JRNL_02-03_0001.webp",
    thumbPath: "public/2/2022_01_03/JRNL_02-03_0001_thumb.webp",
    volume: 2,
    pages: [2, 3],
    date: new Date("2022-01-03T00:00:00.000Z"),
    version: 0,
  },
  {
    uid: "CL_000016",
    path: "public/2/2022_01_03/JRNL_10-11_0001.webp",
    thumbPath: "public/2/2022_01_03/JRNL_10-11_0001_thumb.webp",
    volume: 2,
    pages: [10, 11],
    date: new Date("2022-01-03T00:00:00.000Z"),
    version: 0,
  },
  {
    uid: "CL_000017",
    path: "public/2/2022_01_03/JRNL_12-13_0001.webp",
    thumbPath: "public/2/2022_01_03/JRNL_12-13_0001_thumb.webp",
    volume: 2,
    pages: [12, 13],
    date: new Date("2022-01-03T00:00:00.000Z"),
    version: 0,
  },
  {
    uid: "CL_000018",
    path: "public/1/2021_12_30/JRNL_00-01_0001.webp",
    thumbPath: "public/1/2021_12_30/JRNL_00-01_0001_thumb.webp",
    volume: 1,
    pages: [0, 1],
    date: new Date("2021-12-30T00:00:00.000Z"),
    version: 0,
  },
  {
    uid: "CL_000019",
    path: "public/1/2022_02_01/JRNL_00-01_0001.webp",
    thumbPath: "public/1/2022_02_01/JRNL_00-01_0001_thumb.webp",
    volume: 1,
    pages: [0, 1],
    date: new Date("2022-02-01T00:00:00.000Z"),
    version: 0,
  },
];

const mockState: AppState = {
  scans: mockScans,
  volume: 1,
  highestPage: 13,
  page: 0,
  date: new Date("2022-01-01T00:00:00.000Z"),
  currentScan: mockScans[0],
  manifestPath: "/scans/scan-manifest.json",
};

describe("Page set action", () => {
  test("Sets a valid page number", () => {
    expect(AppReducer(mockInitialState, Action.PAGE_SET(5))).toEqual({
      ...mockInitialState,
      page: 5,
    });
  });

  test("Rejects a negative page number", () => {
    expect(AppReducer(mockInitialState, Action.PAGE_SET(-1))).toEqual(
      mockInitialState
    );
  });

  test("Limits to the highest available page number", () => {
    expect(AppReducer(mockInitialState, Action.PAGE_SET(15))).toEqual({
      ...mockInitialState,
      page: 10,
    });
  });
});

describe("Sequential page nav actions", () => {
  test("Increment page by 2 for double page spread", () => {
    expect(AppReducer(mockInitialState, Action.PAGE_ADD())).toEqual({
      ...mockInitialState,
      page: 2,
    });
  });

  test("Decrement page by 2 for double page spread", () => {
    expect(
      AppReducer(
        {
          ...mockInitialState,
          page: 2,
        },
        Action.PAGE_SUB()
      )
    ).toEqual({
      ...mockInitialState,
      page: 0,
    });
  });

  test("Limit page number decrement to 0", () => {
    expect(AppReducer(mockInitialState, Action.PAGE_SUB())).toEqual(
      mockInitialState
    );
  });

  test("Limit page number increment to highest value", () => {
    expect(
      AppReducer(
        {
          ...mockInitialState,
          page: 10,
        },
        Action.PAGE_ADD()
      )
    ).toEqual({
      ...mockInitialState,
      page: 10,
    });
  });
});

describe("Volume set action", () => {
  test("Sets volume if there are valid scans", () => {
    expect(AppReducer(mockState, Action.VOL_SET(2))).toEqual({
      ...mockState,
      volume: 2,
      currentScan: mockScans[10],
    });
  });

  test("Doesn't set volume if there are no valid scans", () => {
    expect(AppReducer(mockState, Action.VOL_SET(3))).toEqual(mockState);
  });
});

describe("Date set action", () => {
  test("Updates the date and gets the closest scan", () => {
    expect(
      AppReducer(mockState, Action.DATE_SET(new Date("2021-12-29")))
    ).toEqual({
      ...mockState,
      date: new Date("2021-12-29"),
      currentScan: mockScans[18],
    });
  });
});

describe("Sequential date nav action", () => {
  test("Returns unmutated state if currentScan is missing", () => {
    expect(AppReducer(mockInitialState, Action.DATE_SUB())).toEqual(
      mockInitialState
    );
  });
  test("Finds the next scan and updates the date", () => {
    expect(
      AppReducer(
        {
          ...mockState,
          date: new Date("2021-12-29"),
          currentScan: mockScans[18],
        },
        Action.DATE_ADD()
      )
    ).toEqual({ ...mockState, date: mockScans[0].date });
  });
  test("Finds the previous scan and updates the date", () => {
    expect(AppReducer(mockState, Action.DATE_SUB())).toEqual({
      ...mockState,
      date: mockScans[18].date,
      currentScan: mockScans[18],
    });
  });
});

describe("Scan sourcing action", () => {
  test("Sets relevant state values when scan data is imported", () => {
    expect(
      AppReducer(mockInitialState, Action.SCANS_SRC(mockScans))
    ).toMatchSnapshot();
  });
});

describe("Manifest path update action", () => {
  const newPath = "./public/new_manifest.json";
  test("Updates the path to the manifest", () => {
    expect(AppReducer(mockInitialState, Action.MANIFEST_SRC(newPath))).toEqual({
      ...mockInitialState,
      manifestPath: newPath,
    });
  });
});

describe("Unknown actions", () => {
  test("Return unmutated state", () => {
    expect(AppReducer(mockState, { type: "XXXX" })).toEqual(mockState);
  });
});
