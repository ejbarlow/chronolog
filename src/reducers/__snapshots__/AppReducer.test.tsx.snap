// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scan sourcing action Sets relevant state values when scan data is imported 1`] = `
Object {
  "currentScan": Object {
    "date": 2022-01-01T00:00:00.000Z,
    "pages": Array [
      0,
      1,
    ],
    "path": "public/1/2022_01_01/JRNL_00-01_0001.webp",
    "thumbPath": "public/1/2022_01_01/JRNL_00-01_0001_thumb.webp",
    "uid": "CL_000000",
    "version": 0,
    "volume": 1,
  },
  "date": 2022-01-05T19:16:33.000Z,
  "highestPage": 13,
  "manifestPath": "/scans/scan-manifest.json",
  "page": 0,
  "scans": Array [
    Object {
      "date": 2022-01-01T00:00:00.000Z,
      "pages": Array [
        0,
        1,
      ],
      "path": "public/1/2022_01_01/JRNL_00-01_0001.webp",
      "thumbPath": "public/1/2022_01_01/JRNL_00-01_0001_thumb.webp",
      "uid": "CL_000000",
      "version": 0,
      "volume": 1,
    },
    Object {
      "date": 2022-01-01T00:00:00.000Z,
      "pages": Array [
        2,
        3,
      ],
      "path": "public/1/2022_01_01/JRNL_02-03_0001.webp",
      "thumbPath": "public/1/2022_01_01/JRNL_02-03_0001_thumb.webp",
      "uid": "CL_000001",
      "version": 0,
      "volume": 1,
    },
    Object {
      "date": 2022-01-01T00:00:00.000Z,
      "pages": Array [
        4,
        5,
      ],
      "path": "public/1/2022_01_01/JRNL_04-05_0001.webp",
      "thumbPath": "public/1/2022_01_01/JRNL_04-05_0001_thumb.webp",
      "uid": "CL_000002",
      "version": 0,
      "volume": 1,
    },
    Object {
      "date": 2022-01-01T00:00:00.000Z,
      "pages": Array [
        6,
        7,
      ],
      "path": "public/1/2022_01_01/JRNL_06-07_0001.webp",
      "thumbPath": "public/1/2022_01_01/JRNL_06-07_0001_thumb.webp",
      "uid": "CL_000003",
      "version": 0,
      "volume": 1,
    },
    Object {
      "date": 2022-01-01T00:00:00.000Z,
      "pages": Array [
        8,
        9,
      ],
      "path": "public/1/2022_01_01/JRNL_08-09_0001.webp",
      "thumbPath": "public/1/2022_01_01/JRNL_08-09_0001_thumb.webp",
      "uid": "CL_000004",
      "version": 0,
      "volume": 1,
    },
    Object {
      "date": 2022-01-01T00:00:00.000Z,
      "pages": Array [
        10,
        11,
      ],
      "path": "public/1/2022_01_01/JRNL_10-11_0001.webp",
      "thumbPath": "public/1/2022_01_01/JRNL_10-11_0001_thumb.webp",
      "uid": "CL_000005",
      "version": 0,
      "volume": 1,
    },
    Object {
      "date": 2022-01-02T00:00:00.000Z,
      "pages": Array [
        4,
        5,
      ],
      "path": "public/1/2022_01_02/JRNL_04-05_0001.webp",
      "thumbPath": "public/1/2022_01_02/JRNL_04-05_0001_thumb.webp",
      "uid": "CL_000006",
      "version": 0,
      "volume": 1,
    },
    Object {
      "date": 2022-01-02T00:00:00.000Z,
      "pages": Array [
        6,
        7,
      ],
      "path": "public/1/2022_01_02/JRNL_06-07_0001.webp",
      "thumbPath": "public/1/2022_01_02/JRNL_06-07_0001_thumb.webp",
      "uid": "CL_000007",
      "version": 0,
      "volume": 1,
    },
    Object {
      "date": 2022-01-02T00:00:00.000Z,
      "pages": Array [
        10,
        11,
      ],
      "path": "public/1/2022_01_02/JRNL_10-11_0001.webp",
      "thumbPath": "public/1/2022_01_02/JRNL_10-11_0001_thumb.webp",
      "uid": "CL_000008",
      "version": 0,
      "volume": 1,
    },
    Object {
      "date": 2022-01-02T00:00:00.000Z,
      "pages": Array [
        12,
        13,
      ],
      "path": "public/1/2022_01_02/JRNL_12-13_0001.webp",
      "thumbPath": "public/1/2022_01_02/JRNL_12-13_0001_thumb.webp",
      "uid": "CL_000009",
      "version": 0,
      "volume": 1,
    },
    Object {
      "date": 2022-01-02T00:00:00.000Z,
      "pages": Array [
        0,
        1,
      ],
      "path": "public/2/2022_01_02/JRNL_00-01_0001.webp",
      "thumbPath": "public/2/2022_01_02/JRNL_00-01_0001_thumb.webp",
      "uid": "CL_000010",
      "version": 0,
      "volume": 2,
    },
    Object {
      "date": 2022-01-02T00:00:00.000Z,
      "pages": Array [
        2,
        3,
      ],
      "path": "public/2/2022_01_02/JRNL_02-03_0001.webp",
      "thumbPath": "public/2/2022_01_02/JRNL_02-03_0001_thumb.webp",
      "uid": "CL_000011",
      "version": 0,
      "volume": 2,
    },
    Object {
      "date": 2022-01-02T00:00:00.000Z,
      "pages": Array [
        4,
        5,
      ],
      "path": "public/2/2022_01_02/JRNL_04-05_0001.webp",
      "thumbPath": "public/2/2022_01_02/JRNL_04-05_0001_thumb.webp",
      "uid": "CL_000012",
      "version": 0,
      "volume": 2,
    },
    Object {
      "date": 2022-01-02T00:00:00.000Z,
      "pages": Array [
        6,
        7,
      ],
      "path": "public/2/2022_01_02/JRNL_06-07_0001.webp",
      "thumbPath": "public/2/2022_01_02/JRNL_06-07_0001_thumb.webp",
      "uid": "CL_000013",
      "version": 0,
      "volume": 2,
    },
    Object {
      "date": 2022-01-02T00:00:00.000Z,
      "pages": Array [
        8,
        9,
      ],
      "path": "public/2/2022_01_02/JRNL_08-09_0001.webp",
      "thumbPath": "public/2/2022_01_02/JRNL_08-09_0001_thumb.webp",
      "uid": "CL_000014",
      "version": 0,
      "volume": 2,
    },
    Object {
      "date": 2022-01-03T00:00:00.000Z,
      "pages": Array [
        2,
        3,
      ],
      "path": "public/2/2022_01_03/JRNL_02-03_0001.webp",
      "thumbPath": "public/2/2022_01_03/JRNL_02-03_0001_thumb.webp",
      "uid": "CL_000015",
      "version": 0,
      "volume": 2,
    },
    Object {
      "date": 2022-01-03T00:00:00.000Z,
      "pages": Array [
        10,
        11,
      ],
      "path": "public/2/2022_01_03/JRNL_10-11_0001.webp",
      "thumbPath": "public/2/2022_01_03/JRNL_10-11_0001_thumb.webp",
      "uid": "CL_000016",
      "version": 0,
      "volume": 2,
    },
    Object {
      "date": 2022-01-03T00:00:00.000Z,
      "pages": Array [
        12,
        13,
      ],
      "path": "public/2/2022_01_03/JRNL_12-13_0001.webp",
      "thumbPath": "public/2/2022_01_03/JRNL_12-13_0001_thumb.webp",
      "uid": "CL_000017",
      "version": 0,
      "volume": 2,
    },
    Object {
      "date": 2021-12-30T00:00:00.000Z,
      "pages": Array [
        0,
        1,
      ],
      "path": "public/1/2021_12_30/JRNL_00-01_0001.webp",
      "thumbPath": "public/1/2021_12_30/JRNL_00-01_0001_thumb.webp",
      "uid": "CL_000018",
      "version": 0,
      "volume": 1,
    },
    Object {
      "date": 2022-02-01T00:00:00.000Z,
      "pages": Array [
        0,
        1,
      ],
      "path": "public/1/2022_02_01/JRNL_00-01_0001.webp",
      "thumbPath": "public/1/2022_02_01/JRNL_00-01_0001_thumb.webp",
      "uid": "CL_000019",
      "version": 0,
      "volume": 1,
    },
  ],
  "volume": 1,
}
`;
